src/protocol.rs

```diff
@@ -0,0 +1,129 @@
+use serde::de::DeserializeOwned;
```

Comment:
What is the purpose of this module? I thought lspower handled the RPC side of things. Will this module go away once the refactor is complete?

Reply:
This work doesn't do much with the `lspower` side of things. It takes the common helper functions that already exist (as leaky abstractions) and makes them use common `lsp_types` types instead of the in-built types. Without these changes, we'd have to re-implement all of these helper functions for `lspower`, which we'd rather not do. They are ripe for refactor, yes, but we don't want to re-write _everything_ right now.